File: frontend/src/api/eval.ts
import http from './http'

/** 镜像后端 EvalRunOut(app/schemas/eval.py);summary 键与 eval_store.summarize 口径一致 */
export interface EvalRun {
  id: number
  kb_id: number
  kb_name: string | null
  mode: 'retrieval' | 'generation'
  item_count: number
  summary: Record<string, number | null>
  created_at: string
}

/** 镜像后端 EvalItemOut */
export interface EvalItem {
  id: number
  question: string
  expect_doc_ids: number[] | null
  expect_keywords: string[] | null
  answer: string | null
  refused: boolean | null
  hit_at_k: number | null
  mrr: number | null
  keyword_recall: number | null
  faithfulness: number | null
  relevancy: number | null
  reference_score: number | null
}

/** 镜像后端 EvalRunDetailOut */
export interface EvalRunDetail extends EvalRun {
  items: EvalItem[]
  items_truncated: boolean
}

export interface EvalRunQuery {
  kb_id?: number
  mode?: 'retrieval' | 'generation'
  page?: number
  page_size?: number
}

export const evalApi = {
  async listRuns(
    params: EvalRunQuery,
  ): Promise<{ total: number; items: EvalRun[] }> {
    const { data } = await http.get('/eval/runs', { params })
    return data
  },

  async getRun(id: number): Promise<EvalRunDetail> {
    const { data } = await http.get(`/eval/runs/${id}`)
    return data
  },
}

File: backend/app/schemas/eval.py
from datetime import datetime

from pydantic import BaseModel


class EvalRunOut(BaseModel):
    id: int
    kb_id: int
    kb_name: str | None = None  # 装配时联查填入;KB 已删 → None
    mode: str
    item_count: int
    summary: dict
    created_at: datetime

    model_config = {"from_attributes": True}


class EvalItemOut(BaseModel):
    id: int
    question: str
    expect_doc_ids: list | None
    expect_keywords: list | None
    answer: str | None
    refused: bool | None
    hit_at_k: float | None
    mrr: float | None
    keyword_recall: float | None
    faithfulness: float | None
    relevancy: float | None
    reference_score: float | None

    model_config = {"from_attributes": True}


class EvalRunDetailOut(EvalRunOut):
    items: list[EvalItemOut]
    items_truncated: bool
